### PR TITLE
fix: add hiro api-wide rate limiting

### LINKED ISSUE #2750: White sign screen, impossible to sign
I have seen four of these reports from Alexians using the testnet at https://orderbook.alexlab.co I have not been able to reproduce this myself. I have seen it in the past using other dapps though. 

If there is something wrong with the dapp, which causes this, I think it will be great for devs to know what is causing this (could the wallet show a message instead of blinking/blanking). Some users have had success just restarting Chrome and it resolved this. For others that did not help. 

Any ideas, I have asked users to share console details, will add here later if I get any. 

Examples:
image

image

image

image

### PR DESCRIPTION
**Pull request by kyranjamie**
> Try out this version of the Hiro Wallet - download [extension builds](https://github.com/hirosystems/stacks-wallet-web/actions/runs/3631315592).

Updated this feature so that it should be good to merge. Here, we alternate between different rate limiters, to prevent spam of the API.

### CODE REVIEW

**Comment by fbwoolf** [comment]
This appears to be a good idea, should we get this merged sooner than later to possibly help with the white screen issue?

**Comment by markmhendrickson** [comment]
Possibly related white screen issue: https://github.com/hirosystems/stacks-wallet-web/issues/2750

**Comment by fbwoolf** [comment]
@kyranjamie are you going to open this/finish? It seems important for the testnet white screens?

File: src/app/query/stacks/rate-limiter.ts [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+import { RateLimiter } from 'limiter';
+
+import { whenStxChainId } from '@app/common/utils';
+import { useCurrentNetworkState } from '@app/store/networks/networks.hooks';
+
+const oneMinuteInMs = 1_000 * 60;
+
+const hiroStacksMainnetApiLimiter = new RateLimiter({
+  tokensPerInterval: 500,
+  interval: oneMinuteInMs,
```
Comment (edu-stx) [inline src/app/query/stacks/rate-limiter.ts]: Just fyi, also supports `interval: "minute"` which allows getting rid of this var.

**Review by edu-stx — APPROVED** [review]

**Comment by edu-stx** [comment]
If you haven't yet, I can take a look at the failed test

**Comment by kyranjamie** [comment]
> If you haven't yet, I can take a look at the failed test

@edu-stx If you could that'd be great, thanks. I did have a look, it's gotta be something with the request mocking I think, but didn't get to the bottom of how the rate limiting affects this

**Comment by edu-stx** [comment]
Still taking a look at that last failing test.

**Review by edu-stx — APPROVED** [review]

**Comment by edu-stx** [comment]
The other failing test was just flaky, not related to code changed in this PR. Added a small 5s pause in the beforeach hook to allow the app to properly settle before running the tests.

**Review by edu-stx — APPROVED** [review]

**Pull request merged**